ambry-server/src/test/java/com.github.ambry.server/AmbryRequestsTest.java

```diff
@@ -829,6 +945,21 @@ public boolean scheduleNextForCompaction(PartitionId id) {
       return returnValueOfSchedulingCompaction;
     }
 
+    @Override
+    public boolean disableCompactionForBlobStore(PartitionId id) {
+      if (exceptionToThrowOnDisablingCompaction != null) {
+        throw exceptionToThrowOnDisablingCompaction;
+      }
+      compactionDisabledPartitionId = id;
+      return returnValueOfDisablingCompaction;
+    }
+
+    @Override
+    public boolean shutdownBlobStore(PartitionId id) {
+      shutdownPartitionId = id;
```

Comment:
we should also add a test for when this method throws an exception (and handle that in the production code if we don't)